lecture_1/homework/app/controllers/clans_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ClansController < ApplicationController
+
+  def index
+    limit = params.has_key?(:limit) ? params[:limit].to_i : 5
+    clans = Clan.first(limit)
+    render json: clans.to_json(only: clan_attributes), status: :ok
+  end
+
+  def create
+    clan = Clan.new(clan_params)
+
+    if clan.save
+      render json: clan.to_json(only: clan_attributes), status: 201
+    else
+      render json: { errors: clan.errors.messages }, status: 422
+    end
+  end
+
+  def dead
```

Comment:
To tu przy okazji odpowiadając na pytanie, przy projektowaniu restowego api, najlepiej nie tworzyć takich `nierestowych` akcji, tylko obsługiwać je jakoś w odpowiedniej restowej akcji(dostępne w CRUD to show, index, create, update, destroy) albo wydzielić do nowego zagnieżdżonego kontrolera, w naszym przypadku 100% full super restowo ścieżki by wyglądały tak:

```ruby
resources :clans, param: :name  do
  resources :samurais do
    resources :alive, only: %i[index]
    resources :dead, only: %i[index]
  end
end
```

Oczywiście staramy się unikać zagnieżdżeń więc to póki co samo obsługiwanie 2 query params w indexie by wystarczyło, ale jeśli mielibyśmy np. inne akcje `update` dla żywych i martwych samurai, to taką architekturę dużo łatwiej się rozbudowuje i ładniej dzieli tą logikę :D

Wieloraki sposób dostępu jest dobrą praktyką, jeśli te wielorakie ścieżki dostępu nie są identyczne.

Reply:
Ooooo, mega dziękuje za to wytłumaczenie, to mi sporo rozjaśniło w kontekście REST'a

Reply:
Polecam też takie pytania na slacku, gdzie więcej osób się pali żeby wam pomóc :)